frappe/model/naming.py

```diff
@@ -235,13 +235,21 @@ def set_naming_from_document_naming_rule(doc):
 	if doc.doctype in log_types:
 		return
 
-	# ignore_ddl if naming is not yet bootstrapped
-	for d in frappe.get_all(
-		"Document Naming Rule",
-		dict(document_type=doc.doctype, disabled=0),
-		order_by="priority desc",
-		ignore_ddl=True,
-	):
+	def _get_document_naming_rule():
```

Comment:
There should be cache eviction for this whenever doc naming rule is updated/inserted.